test/lifecycle_test.dart

```diff
@@ -175,6 +175,29 @@ main() {
 
         sharedTypeTests(testSnapshotType);
       });
+
+      test('triggers error lifecycle events when an error is thrown', () {
+        var mountNode = new DivElement();
+        var renderedInstance = react_dom.render(components2.SetStateTest({}), mountNode);
+        LifecycleTestHelper component = getDartComponent(renderedInstance);
+        LifecycleTestHelper.staticLifecycleCalls.clear();
+        component.setState({"shouldThrow": true});
+
+        expect(component.lifecycleCalls, containsAllInOrder(['getDerivedStateFromError', 'componentDidCatch']));
```

Comment:
Is there a reason `containsAllInOrder` is used here? I think `contains` would give us more test coverage, and assert that methods aren't called when we don't expect them to.